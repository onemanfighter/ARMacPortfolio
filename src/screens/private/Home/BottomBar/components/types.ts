export enum ProgramType {
  FINDER = 'finder',
  CHROME = 'chrome',
  VSCODE = 'vscode',
  TERMINAL = 'terminal',
  SPOTIFY = 'spotify',
  GITHUB = 'github',
  SETTINGS = 'settings',
}

export type IconButtonProps = {
  type: ProgramType;
};
