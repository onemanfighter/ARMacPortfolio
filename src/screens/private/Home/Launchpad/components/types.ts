import { ProgramType } from '@processStore';

export type ProgramButtonProps = {
  name: string;
  icon: string;
  isActive: boolean;
  onClickHandler: () => void;
};
