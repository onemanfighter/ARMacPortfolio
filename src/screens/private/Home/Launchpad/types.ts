import { ProgramType } from '@processStore';

export type LaunchpadProps = {};

export type ProgramData = {
  name: string;
  icon: string;
  programType: ProgramType;
};
