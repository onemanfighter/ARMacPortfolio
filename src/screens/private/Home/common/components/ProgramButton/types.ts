import { ProgramType } from '@processStore';

export type ProgramButtonProps = {
  name: string;
  icon: string;
  isActive: boolean;
  variant?: 'default' | 'bottomBar';
  onClickHandler: () => void;
  size?: number;
};
