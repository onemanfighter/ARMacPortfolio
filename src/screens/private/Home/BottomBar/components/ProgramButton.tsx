import { Box, IconButton, Img, Tooltip } from '@chakra-ui/react';
import { IconButtonProps } from './types';
import PROGRAMS from './constants';
import { displayDockSelector, settingsStore, useShallow } from '@settingsStore';
import { getIconSize } from './utils';
import { Draggable } from 'react-beautiful-dnd';

const ProgramButton = ({ isActive, type, index, id }: IconButtonProps) => {
  const { dockSize } = settingsStore(useShallow(displayDockSelector));
  const { name, icon } = PROGRAMS[type] ?? { name: 'Unknown', icon: 'unknown' };
  const size = getIconSize(dockSize);
  return (
    <Draggable draggableId={id} index={index} key={name + '-button'}>
      {(provided) => (
        <Box
          ref={provided.innerRef}
          {...provided.draggableProps}
          {...provided.dragHandleProps}
          display={'flex'}
          flexDir={'column'}
          alignItems={'center'}
        >
          <Tooltip
            borderRadius={6}
            label={name}
            hasArrow
            arrowSize={15}
            top={-4}
          >
            <Img
              boxSize={size}
              src={icon}
              height={size}
              width={size}
              alt={name}
              _active={{
                filter: 'brightness(0.6)',
              }}
              loading="lazy"
            />
          </Tooltip>
          {isActive ? (
            <Box boxSize={1} bgColor={'white'} borderRadius={'full'} />
          ) : null}
        </Box>
      )}
    </Draggable>
  );
};

export default ProgramButton;
