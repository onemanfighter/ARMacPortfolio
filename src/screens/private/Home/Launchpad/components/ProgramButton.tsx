import { Box, Img, Text, Tooltip } from '@chakra-ui/react';
import { ProgramButtonProps } from './types';
import {
  darkModeColorSelector,
  settingsStore,
  useShallow,
} from '@settingsStore';
import { APP_ICON_SIZE } from './constants';

const ProgramButton = ({
  icon,
  name,
  isActive,
  onClickHandler,
}: ProgramButtonProps) => {
  const { bottomBarActiveDot, BottomBarTooltipBgColor, textColor } =
    settingsStore(useShallow(darkModeColorSelector));
  return (
    <Box
      key={name + '-button'}
      aria-label={`program-button-${name}`}
      display={'flex'}
      _hover={{
        cursor: 'pointer',
      }}
      _active={{
        cursor: 'pointer',
      }}
      flexDir={'column'}
      alignItems={'center'}
      onClick={onClickHandler}
    >
      <Img
        boxSize={APP_ICON_SIZE}
        src={icon}
        height={APP_ICON_SIZE}
        width={APP_ICON_SIZE}
        alt={name}
        _active={{
          filter: 'brightness(0.6)',
        }}
        loading="lazy"
      />
      <Text color={textColor} fontSize={'sm'}>
        {name}
      </Text>
      {isActive ? (
        <Box boxSize={1} bgColor={bottomBarActiveDot} borderRadius={'full'} />
      ) : null}
    </Box>
  );
};

export default ProgramButton;
