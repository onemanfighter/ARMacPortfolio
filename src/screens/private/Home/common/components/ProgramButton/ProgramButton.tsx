import { Box, Img, Text } from '@chakra-ui/react';
import { ProgramButtonProps } from './types';
import {
  darkModeColorSelector,
  displayDockSelector,
  settingsStore,
  useShallow,
} from '@settingsStore';
import { APP_ICON_SIZE } from './constants';
import { getIconSize } from './utils';

const ProgramButton = ({
  icon,
  name,
  isActive,
  variant = 'default',
  onClickHandler,
}: ProgramButtonProps) => {
  const { dockSize } = settingsStore(useShallow(displayDockSelector));
  const size = variant === 'bottomBar' ? getIconSize(dockSize) : APP_ICON_SIZE;
  const { bottomBarActiveDot, textColor } = settingsStore(
    useShallow(darkModeColorSelector),
  );
  return (
    <Box
      key={name + '-button'}
      aria-label={`program-button-${name}`}
      display={'flex'}
      _hover={{
        cursor: 'pointer',
      }}
      _active={{
        cursor: 'pointer',
      }}
      flexDir={'column'}
      alignItems={'center'}
      onClick={onClickHandler}
    >
      <Img
        boxSize={size}
        src={icon}
        height={size}
        width={size}
        alt={name}
        _active={{
          filter: 'brightness(0.6)',
        }}
        loading="lazy"
      />
      {variant === 'default' ? (
        <Text color={textColor} fontSize={'sm'}>
          {name}
        </Text>
      ) : null}
      {isActive ? (
        <Box boxSize={1} bgColor={bottomBarActiveDot} borderRadius={'full'} />
      ) : null}
    </Box>
  );
};

export default ProgramButton;
