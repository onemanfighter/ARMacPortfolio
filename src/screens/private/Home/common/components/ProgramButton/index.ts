export { default as ProgramButton } from './ProgramButton';
export { getIconSize } from './utils';
