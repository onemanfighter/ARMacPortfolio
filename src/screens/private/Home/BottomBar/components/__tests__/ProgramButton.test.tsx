import { render } from '@testing-library/react';
import ProgramButton from '../ProgramButton';
import { ProgramType } from '../types';
import { DragDropContext, Droppable } from 'react-beautiful-dnd';

describe('Program button', () => {
  it('should render finder correctly to match snapshot', () => {
    const { container } = render(
      <DragDropContext onDragEnd={jest.fn()}>
        <Droppable droppableId="list" direction="horizontal">
          {() => (
            <ProgramButton
              type={ProgramType.FINDER}
              id={'Finder'}
              index={0}
              isActive={false}
            />
          )}
        </Droppable>
      </DragDropContext>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render chrome correctly to match snapshot', () => {
    const { container } = render(
      <DragDropContext onDragEnd={jest.fn()}>
        <Droppable droppableId="list" direction="horizontal">
          {() => (
            <ProgramButton
              type={ProgramType.CHROME}
              id={'Chrome'}
              index={0}
              isActive={false}
            />
          )}
        </Droppable>
      </DragDropContext>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render vscode correctly to match snapshot', () => {
    const { container } = render(
      <DragDropContext onDragEnd={jest.fn()}>
        <Droppable droppableId="list" direction="horizontal">
          {() => (
            <ProgramButton
              type={ProgramType.VSCODE}
              id={'VsCode'}
              index={0}
              isActive={false}
            />
          )}
        </Droppable>
      </DragDropContext>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render terminal correctly to match snapshot', () => {
    const { container } = render(
      <DragDropContext onDragEnd={jest.fn()}>
        <Droppable droppableId="list" direction="horizontal">
          {() => (
            <ProgramButton
              type={ProgramType.TERMINAL}
              id={'Terminal'}
              index={0}
              isActive={false}
            />
          )}
        </Droppable>
      </DragDropContext>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render spotify correctly to match snapshot', () => {
    const { container } = render(
      <DragDropContext onDragEnd={jest.fn()}>
        <Droppable droppableId="list" direction="horizontal">
          {() => (
            <ProgramButton
              type={ProgramType.SPOTIFY}
              id={'Spotify'}
              index={0}
              isActive={false}
            />
          )}
        </Droppable>
      </DragDropContext>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render github correctly to match snapshot', () => {
    const { container } = render(
      <DragDropContext onDragEnd={jest.fn()}>
        <Droppable droppableId="list" direction="horizontal">
          {() => (
            <ProgramButton
              type={ProgramType.GITHUB}
              id={'Github'}
              index={0}
              isActive={false}
            />
          )}
        </Droppable>
      </DragDropContext>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render settings correctly to match snapshot', () => {
    const { container } = render(
      <DragDropContext onDragEnd={jest.fn()}>
        <Droppable droppableId="list" direction="horizontal">
          {() => (
            <ProgramButton
              type={ProgramType.SETTINGS}
              id={'Settings'}
              index={0}
              isActive={false}
            />
          )}
        </Droppable>
      </DragDropContext>,
    );

    expect(container).toMatchSnapshot();
  });
});
