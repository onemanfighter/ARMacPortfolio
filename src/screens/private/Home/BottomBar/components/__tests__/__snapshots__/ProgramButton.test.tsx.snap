// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Program button should render chrome correctly to match snapshot 1`] = `
<div>
  <div
    aria-describedby="rbd-hidden-text-1-hidden-text-3"
    class="css-16co9uo"
    data-rbd-drag-handle-context-id="1"
    data-rbd-drag-handle-draggable-id="Chrome"
    data-rbd-draggable-context-id="1"
    data-rbd-draggable-id="Chrome"
    draggable="false"
    role="button"
    tabindex="0"
  >
    <img
      alt="Chrome"
      class="chakra-image css-h67rfo"
      loading="lazy"
      src=""
    />
  </div>
</div>
`;

exports[`Program button should render finder correctly to match snapshot 1`] = `
<div>
  <div
    aria-describedby="rbd-hidden-text-0-hidden-text-0"
    class="css-16co9uo"
    data-rbd-drag-handle-context-id="0"
    data-rbd-drag-handle-draggable-id="Finder"
    data-rbd-draggable-context-id="0"
    data-rbd-draggable-id="Finder"
    draggable="false"
    role="button"
    tabindex="0"
  >
    <img
      alt="Finder"
      class="chakra-image css-h67rfo"
      loading="lazy"
      src=""
    />
  </div>
</div>
`;

exports[`Program button should render github correctly to match snapshot 1`] = `
<div>
  <div
    aria-describedby="rbd-hidden-text-5-hidden-text-15"
    class="css-16co9uo"
    data-rbd-drag-handle-context-id="5"
    data-rbd-drag-handle-draggable-id="Github"
    data-rbd-draggable-context-id="5"
    data-rbd-draggable-id="Github"
    draggable="false"
    role="button"
    tabindex="0"
  >
    <img
      alt="Github"
      class="chakra-image css-h67rfo"
      loading="lazy"
      src=""
    />
  </div>
</div>
`;

exports[`Program button should render settings correctly to match snapshot 1`] = `
<div>
  <div
    aria-describedby="rbd-hidden-text-6-hidden-text-18"
    class="css-16co9uo"
    data-rbd-drag-handle-context-id="6"
    data-rbd-drag-handle-draggable-id="Settings"
    data-rbd-draggable-context-id="6"
    data-rbd-draggable-id="Settings"
    draggable="false"
    role="button"
    tabindex="0"
  >
    <img
      alt="Settings"
      class="chakra-image css-h67rfo"
      loading="lazy"
      src=""
    />
  </div>
</div>
`;

exports[`Program button should render spotify correctly to match snapshot 1`] = `
<div>
  <div
    aria-describedby="rbd-hidden-text-4-hidden-text-12"
    class="css-16co9uo"
    data-rbd-drag-handle-context-id="4"
    data-rbd-drag-handle-draggable-id="Spotify"
    data-rbd-draggable-context-id="4"
    data-rbd-draggable-id="Spotify"
    draggable="false"
    role="button"
    tabindex="0"
  >
    <img
      alt="Spotify"
      class="chakra-image css-h67rfo"
      loading="lazy"
      src=""
    />
  </div>
</div>
`;

exports[`Program button should render terminal correctly to match snapshot 1`] = `
<div>
  <div
    aria-describedby="rbd-hidden-text-3-hidden-text-9"
    class="css-16co9uo"
    data-rbd-drag-handle-context-id="3"
    data-rbd-drag-handle-draggable-id="Terminal"
    data-rbd-draggable-context-id="3"
    data-rbd-draggable-id="Terminal"
    draggable="false"
    role="button"
    tabindex="0"
  >
    <img
      alt="Terminal"
      class="chakra-image css-h67rfo"
      loading="lazy"
      src=""
    />
  </div>
</div>
`;

exports[`Program button should render vscode correctly to match snapshot 1`] = `
<div>
  <div
    aria-describedby="rbd-hidden-text-2-hidden-text-6"
    class="css-16co9uo"
    data-rbd-drag-handle-context-id="2"
    data-rbd-drag-handle-draggable-id="VsCode"
    data-rbd-draggable-context-id="2"
    data-rbd-draggable-id="VsCode"
    draggable="false"
    role="button"
    tabindex="0"
  >
    <img
      alt="VsCode"
      class="chakra-image css-h67rfo"
      loading="lazy"
      src=""
    />
  </div>
</div>
`;
